llvm/test/CodeGen/AArch64/veclib-llvm.modf.ll

```diff
@@ -0,0 +1,57 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --filter "(bl|ptrue)" --version 5
+; RUN: llc -mtriple=aarch64-gnu-linux -mattr=+neon,+sve -vector-library=sleefgnuabi < %s | FileCheck %s -check-prefix=SLEEF
+; RUN: llc -mtriple=aarch64-gnu-linux -mattr=+neon,+sve -vector-library=ArmPL < %s | FileCheck %s -check-prefix=ARMPL
+
+define <4 x float> @test_modf_v4f32(<4 x float> %x, ptr %out_integral) {
+; SLEEF-LABEL: test_modf_v4f32:
+; SLEEF:    bl _ZGVnN4vl4_modff
+;
+; ARMPL-LABEL: test_modf_v4f32:
+; ARMPL:    bl armpl_vmodfq_f32
```

Comment:
I'd rather see the full output here because the code goes to some trouble to ensure the IR stores are merged with the library call and so it seems prudent to verify that happens.  Perhaps worth having some negative tests (e.g. where %out_integral is loaded from after the call but before the IR store) unless that scenario is already covered elsewhere?

Reply:
Sure, I've removed the filtering. I'll add a few negative tests later on (I think there's some already for sincos/frexp for the store folding, but possibly not testing vectors too).

Reply:
Awesome. Thanks.

Reply:
I've added `modf_store_merging_load_before_store()` and you can see the behaviour is correct (the load is before the call, and the value is saved around the call).